### DISCUSSION TITLE
Is it possible to have a kinematic body penetrations against static bodies resolved?

### DISCUSSION THREAD

**Discussion by bobbydigitales** [jrouwe/JoltPhysics#699] [Q&A]
Hi,

I'm trying to set up a simple player controller that does the following:
1) The player is a simple kinematic box shape
2) The game moves the player body each tick using MoveKinematic
3) I tick Jolt, for dynamic bodies, they are pushed by the player, for static bodies, the player doesn't penetrate them but slides along them.
4) I get ContactListener callbacks for all added/persisted/removed contacts on the player.

I have a lot of this working however:

1. When I set the player to Kinematic, it pushes dynamic bodies around, but it is not stopped by static bodies and I don't get callbacks via my ContactListener

3. When I set the player body to be a sensor, I get callbacks, but dynamic bodies are not affected by the player body, and the player body is not stopped by static bodies.

Is it possible to have my cake and eat it, where the kinematic player body affects dynamic bodies, but is stopped by static bodies, and I get ContactListener callbacks for all contacts on the player body without having to add an additional sensor body to the player?

Thanks!

Rob

**Comment by jrouwe** [comment]
> Is it possible to have my cake and eat it, where the kinematic player body affects dynamic bodies, but is stopped by static bodies, and I get ContactListener callbacks for all contacts on the player body without having to add an additional sensor body to the player?

No this is not possible (or at least not within Jolt). The definition of a kinematic object is that is has infinite mass. This means that nothing can push it away. A static object also has infinite mass, so if we were to allow two infinitely heavy objects to interact with each other it would be undefined what would happen.

What you can do is to perform an additional collision check and update the velocity of the kinematic body to stop at a static object yourself.

Making player controllers is difficult stuff though, I would advice you to take a look at ```CharacterVirtual```.
